packaging/opensuse/snapd.spec

```diff
@@ -143,18 +156,37 @@ updated as soon as new upstream releases are made and is designed to live in
 the system:snappy repository.
 
 %prep
-%setup -q
-
-# Set the version that is compiled into the various executables
+# NOTE: Instead of using setup -q we are unpacking a subdirectory of the source
+# tarball into a directory that is automatically on the future GOPATH. This
+# means that while go doesn't care at all the current working directory is not
+# the top-level directory of the source tarball which some people may find
+# unusual.
+
+# Create indigo compatible build layout.
+mkdir -p %{indigo_srcdir}
+tar -axf %{_sourcedir}/%{name}_%{version}.vendor.tar.xz --strip-components=1 -C %{indigo_srcdir}
+
+# Patch the source in the place it got extracted to.
+pushd %{indigo_srcdir}
+# Add patch0 -p1 ... as appropriate here.
+popd
+
+# Set the version that is compiled into the various executables/
+pushd %{indigo_srcdir}
 ./mkversion.sh %{version}-%{release}
+popd
 
-# Generate autotools build system files
-cd cmd && autoreconf -i -f
+# Sanity check, ensure that systemd system generator directory is in agreement between the build system and packaging.
+if [ "$(pkg-config --variable=systemdsystemgeneratordir systemd)" != "%{_systemdgeneratordir}" ]; then
+  echo "pkg-confing and rpm macros disagree about the location of systemd system generator directory"
```

Comment:
Heh, does this really happen?

Reply:
Well, that's the reason I spent some much time on this!

Our default in configure.ac has some value that happens to be used if systemd.pc is not present. It is a silent bug that just happens. To ensure that we don't miss this the next time I added this check.

Reply:
Feels like this should be a bug against systemd packaging or `systemd-rpm-macros`

Reply:
The bug is on our side, we should not guesstimate any directories.